uaclient/clouds/aws.py

```diff
@@ -43,10 +42,33 @@ def _get_imds_url_response(self):
     @property  # type: ignore
     @util.retry(exceptions.CloudMetadataError, retry_sleeps=[0.5, 1, 1])
     def identity_doc(self) -> Dict[str, Any]:
-        imds_url_response = self._get_imds_url_response()
+        headers = self._request_imds_v2_token_headers()
+        url = IMDS_URL.format(self._ip_address)
+        imds_url_response = self._get_imds_url_response(url, headers=headers)
         secret_manager.secrets.add_secret(imds_url_response)
         return {"pkcs7": imds_url_response}
 
+    @util.retry(exceptions.CloudMetadataError, retry_sleeps=[0.5, 1, 1])
+    def _get_ii_doc(self) -> Dict:
```

Comment:
also could you add docstrings to this function and the below `is_likely_pro` function?

Reply:
Done.